content/posts/cloudbees-core-on-kubernetes-best-practices/core-psp.md

```diff
@@ -0,0 +1,234 @@
+
+## OOTB and Documented Security Best Practices for CloudBees Core v2 on Kubernetes
+There are [numerous articles](https://rancher.com/blog/2019/2019-01-17-101-more-kubernetes-security-best-practices/) [on security best practices](https://www.twistlock.com/2019/06/06/5-kubernetes-security-best-practices/) for Kubernetes (to include [this one published on the CNCF blog site](https://www.cncf.io/blog/2019/01/14/9-kubernetes-security-best-practices-everyone-must-follow/)). Many of these articles include similar best practices and most, if not all, apply to running Core v2 on Kubernetes. Some of these best practices are inherent in CloudBees' documented install of Core v2 on Kubernetes, while others are documented best practices and are recommended next steps after your initial Core v2 installation. 
+
+Before we take a look at the best practices that aren't necessarily covered by the CloudBees reference architectures and best practice documentation, I will provide a quick overview of what is already available with an OOTB Core v2 install and highlight some CloudBees documentation that speaks to other best practices for running Core v2 on Kubernetes more securely.
+
+### Enable Role-Based Access Control (RBAC)
+Although you can certainly install Core v2 on Kubernetes without RBAC enabled - the CloudBees install for Core v2 comes with RBAC pre-configured. Running Kubernetes with RBAC enabled is typically the default (it is for all the major cloud providers) and is always a recommended security setting.
+
+### Use Namespaces to Establish Security Boundaries & Separate Sensitive Workloads
+CloudBees recommends that you create a `namespace` specifically for Core v2 as part of the install. CloudBees also recommends establishing boundaries between your CloudBees Jenkins masters and agent workloads by [setting up distinct node pools](https://go.cloudbees.com/docs/cloudbees-core/cloud-install-guide/gke-install/#_distinct_node_pools) using [taints and tolerations](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/) and [assigning pods to specific node pools with node selectors](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/).
+
+### Create and Define Cluster Network Policies
+Although CloudBees doesn't provide specific Kubernetes Network Policies, CloudBees does recommend using them and [provides documentation for setting up a private and encrypted network for AWS EKS](https://go.cloudbees.com/docs/cloudbees-core/cloud-reference-architecture/ra-for-eks/#_setting_up_a_private_and_encrypted_network).
+
+### Run a Cluster-wide Pod Security Policy
+At the time of this post, this is one component that is not documented as part of the CloudBees installation guides for Core v2 on Kubernetes and will be the focus of the rest of this post.
+
+## Why should you use Pod Security Policies?
```

Comment:
I think it should be interesting to link this to the privilege container execution that you explained in the previous post. Then also can be related with your later configuration of the PSP showing the `runAsUser` configuration... just an idea

Reply:
Done